systemtest/src/main/java/io/strimzi/systemtest/kafkaclients/externalClients/TracingExternalKafkaClient.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.systemtest.kafkaclients.externalClients;
+
+import io.strimzi.systemtest.kafkaclients.AbstractKafkaClient;
+import io.strimzi.systemtest.kafkaclients.IKafkaClientOperations;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.util.concurrent.Future;
+
+/**
+ * The TracingKafkaClient for sending and receiving messages using tracing properties.
+ * The client is using an external listeners.
+ */
+public class TracingExternalKafkaClient extends AbstractKafkaClient implements IKafkaClientOperations<Future<Integer>> {
+
+    private static final Logger LOGGER = LogManager.getLogger(TracingExternalKafkaClient.class);
+    private String serviceName;
+
+    public static class Builder extends AbstractKafkaClient.Builder<TracingExternalKafkaClient.Builder> {
+
+        private String serviceName;
+
+        public Builder withServiceName(String serviceName) {
+
+            this.serviceName = serviceName;
+            return self();
+        }
+
+        @Override
+        public TracingExternalKafkaClient build() {
+
+            return new TracingExternalKafkaClient(this);
+        }
+
+        @Override
+        protected TracingExternalKafkaClient.Builder self() {
+            return this;
+        }
+    }
+
+    private TracingExternalKafkaClient(TracingExternalKafkaClient.Builder builder) {
+
+        super(builder);
+        serviceName = builder.serviceName;
+    }
+
+    @Override
+    public Future<Integer> sendMessagesPlain(long timeoutMs) {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public Future<Integer> sendMessagesTls(long timeoutMs) {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public Future<Integer> receiveMessagesPlain(long timeoutMs) {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public Future<Integer> receiveMessagesTls(long timeoutMs) {
+        throw new UnsupportedOperationException();
+    }
```

Comment:
Something is wrong if you're implementing an interface with all methods throwing `UnsupportedOperationException`. I'm _guessing_ you've had to do this because you wanted a different builder for the tracing case? I think you need to question why your builder has to be an inner class of some `AbstractKafkaClient` subclass.

Reply:
Practically, this should be implemented in a few weeks. It will be the Tracing clients using these methods and yeah.

Reply:
OK. A `// TODO` or exception message to that effect is always helpful to show the intent to come back and address it.